continuous-integration/reference-architectures/README.md

```diff
@@ -1,3 +1,181 @@
+to the CI philosophy.  
+- **Report**  
+At the core of CI is reporting, both notifications and functional reporting.  
+_Notifications:_ Developers are notified on successful completion of builds or failures. 
+Notifications also trigger subsequent automation steps and systems.  
+_Functional Reporting:_ (compile status, test results) Reports are a built-in feature in many CI 
+systems, but extended reports can be generated also to track such items as test results locally or 
+in connected systems. The Jenkins build server, for instance, maintains build logs that tie commit 
+history and compile results to every single execution. Similarly, results from tests can be reported 
+using extended tooling to servers such as [Gurock TestRail](https://www.gurock.com/testrail/) 
+to retain test runs from every build. Build logs typically span all aspects of the 
+compile-release-publish workflow, from code checkout to publishing compiled binaries. 
+- **Package**  
+The result of code compilation must be deliverable as manageable objects, and that is the purpose 
+of the packaging phase. One or more compiled binaries may exist, as well as a set of properties or 
+other application-required configurations. These are bundled into one or more collections typically 
+compressed using one of a number of common formats, such as ZIP, JAR or TAR. In practice many build 
+frameworks, such as [Apache Maven](https://maven.apache.org/), package automatically. This step 
+allows the CI system to bundle, protect and lock down files for distribution and reproducibility.
+- **Publish**  
+The results of the compile-release-publish process are packaged binaries. These are shared with 
+development and devops communities by publishing the files to one or more artifact repositories, 
+such as [JFrog Artifactory](https://jfrog.com/artifactory/). Packaged binaries are versioned, 
+checksummed and do not change after release -- core features of reliable and reproducible CI 
+software delivery.
+
+#### Continuous Deployment System
+- **Orchestrate** (optional)  
+The orchestration process downloads files and prepares target servers to launch. It moves compiled 
+binaries and properties into place and configures required services on destination systems. Depending 
+on complexity of the final product this step is optional.
+- **Release** (optional)  
+In CI implementations releasing product typically starts and announces applications on a development 
+server used for extended functionality testing. An extension of this, Continuous Delivery (CD), 
+automates functionality testing and deploys to multiple [release runway](https://www.techtarget.com/searchsoftwarequality/definition/staging-environment) 
+destinations, including potentially directly to production. Releasing compiled product is optional 
+depending on intended use cases (executable vs. API library) or whether further testing is desired.
+
+### Actors and Automation
```

Comment:
You've used the term "actors" in the title but not used it anywhere in the description. Suggestion to rephrase.

Reply:
This has been addressed with text that explains the word usage. See next commit.